### PR TITLE
Pre post increment ops renamed

### PR DESCRIPTION
**Pull request by zwimer**
This is a feature add that also fixes the same bugs that https://github.com/RosettaCommons/binder/pull/237 fixes (as the code introduced in https://github.com/RosettaCommons/binder/pull/237 made it easy to add this feature). It addresses https://github.com/RosettaCommons/binder/issues/226

### CODE REVIEW

**Comment by zwimer** [comment]
As mentioned in https://github.com/RosettaCommons/binder/pull/237 there are a few possible options we can do:

1. Close this and post/pre increment both named `plus_plus`, with post requiring a integer dummy argument (This makes it harder to tell `++a` from `a++` apart)
2. Merge this and rename define `pre_increment()` and `post_increment(int)` (This is not backwards compatible)
3. Improve this and define `pre_increment()` and `post_increment()` (This would remove the ability to control that dummy integer in the rare case someone decided to use it for some reason)
4. Do either 2 or 3, but also keep `plus_plus` for compatibility (This might add overhead)
5. Add a CLI flag to decide between options

**Comment by zwimer** [comment]
Should I do anything to get this PR moving?

File: source/function.cpp [hunk 1]
```diff
@@ -73,8 +73,8 @@ string cpp_python_operator(const FunctionDecl & F) {
 		{"operator!=", {"__ne__"}}, //
 		{"operator[]", {"__getitem__"}}, //
 		{"operator=", {"assign"}}, //
-		{"operator++", {"plus_plus"}}, //
-		{"operator--", {"minus_minus"}}, //
+		{"operator++", {"pre_increment", "pre_increment"}}, //
```
Comment (lyskov) [inline source/function.cpp:76]: probably `{"operator++", {"pre_increment", "post_increment"}},` was intended

**Review by lyskov — COMMENTED** [review]
Sorry if my previous comment on was a bit cryptic. The main issue with merging this is backward compatibility with already generate code. However i just checked and it looks like there is only a few cases where these was uses in my project so i am open on merging this. Please see minor comment below that needs to be address before this is merged. Thanks,

**Review by lyskov — APPROVED** [review]
LGTM, - thank you for putting this together @zwimer !

**Comment by lyskov** [comment]
@aleaverfay Hi Andrew! Sorry to hear that these changes broke your code! These change have not break any of PyRosetta unit tests so I am guessing that it could be something specific to area of code you are working with. Never the less let's see if we can find a good solution that works for you!

I am a bit skeptical about "double bindings" since it will just prolong the issue (and then we will have one more API function to take care of) (and currently Binder does not have such functionality). The code that got broken, - is that something that you will be OK/easy/have-permission to modify? If so, have you considered the following workaround which should have ~zero performance impact:

During import phase, we can check if particular class already have `plus_plus` defined and if not just link its implementation to `pre_increment`/`post_increment`, like this:

```Python
...
import aaaa
if not hasattr(aaaa.A, 'plus_plus'): aaaa.A.plus_plus = aaaa.A.pre_increment
```

-- will something like this work for you? Thanks,

**Comment by lyskov** [comment]
@aleaverfay Hi Andrew! Sorry to hear that these changes broke your code! These change have not break any of PyRosetta unit tests (and I have not heard so far from other users regarding this change) so I am guessing that it could be something specific to area of code you are working with. Never the less let's see if we can find a good solution that works for you!

I am a bit skeptical about "double bindings" since it will just prolong the issue (and then we will have one more API function to take care of) (and currently Binder does not have such functionality). The code that got broken, - is that something that you will be OK/easy/have-permission to modify? If so, have you considered the following workaround which should have ~zero performance impact:

During import phase, we can check if particular class already have `plus_plus` defined and if not just link its implementation to `pre_increment`/`post_increment`, like this:

```Python
...
import aaaa
if not hasattr(aaaa.A, 'plus_plus'): aaaa.A.plus_plus = aaaa.A.pre_increment
```

-- will something like this work for you? Thanks,

**Pull request merged**